### PR TITLE
feat: add register endpoint

### LINKED ISSUE #3: Add register endpoint
Add a register endpoint to create temporary account in case of mobile number auth or social signin
1. For mobile auth parameter to accept name and mobile number
2. For social signin parameter to accept name and email

### PR DESCRIPTION
**Pull request by anujakiran**
Add register endpoint

### CODE REVIEW

File: src/helper/responseHelper.ts [hunk 1]
```diff
@@ -3,10 +3,21 @@ import { Response } from 'express';
 export const responseHelperSuccess = (
   res: Response,
   statusCode: number,
-  result: Record<string, string>,
+  result?: Record<string, unknown>,
```
Comment (Kiransukumaran) [inline src/helper/responseHelper.ts]: Remove this unknown

File: src/router/accountRouter.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Router } from 'express';
+
+import { createAccount } from '../controller/accountController';
+import { validateRequest } from '../helper/validateResultHelper';
+import { createAccountBodyValidation } from '../middleware/accountMiddleware';
+
+const AccountRouter = Router();
+
+AccountRouter.post(
+  '/accounts',
```
Comment (Kiransukumaran) [inline src/router/accountRouter.ts:10]: /account

File: src/router/accountRouter.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Router } from 'express';
+
+import { createAccount } from '../controller/accountController';
+import { validateRequest } from '../helper/validateResultHelper';
+import { createAccountBodyValidation } from '../middleware/accountMiddleware';
+
+const AccountRouter = Router();
```
Comment (Kiransukumaran) [inline src/router/accountRouter.ts]: accountRouter

File: src/services/accountService.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import User from '../db/models/user';
+import { BaseRepository } from '../repositories/baseRepository';
+import { ResponseData } from '../types/common';
+
+export class AccountService {
+  constructor(private accountRepo: BaseRepository<User>) {}
+
+  async createAccountService(req: Request): Promise<ResponseData> {
+    try {
+      if (req.query.mobileAuth) {
```
Comment (Kiransukumaran) [inline src/services/accountService.ts]: Determine this with body params

File: src/services/accountService.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            statusCode: 400,
+            error: 'email is required for social login',
+          };
+        }
+        const result = await this.accountRepo.create(req.body, {});
+        return {
+          statusCode: 201,
+          result,
+        };
+      }
```
Comment (Kiransukumaran) [inline src/services/accountService.ts]: Code duplication here refactor this to make more concise

File: src/services/accountService.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    } catch (error) {
+      if (error instanceof UniqueConstraintError) {
+        return {
+          statusCode: 303,
```
Comment (Kiransukumaran) [inline src/services/accountService.ts:28]: Can you add a logger and add logs in case of error for debugging

File: src/types/common.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+type ResponseData = {
```
Comment (Kiransukumaran) [inline src/types/common.ts:1]: Nit but if it's ok please split it into success response and error response

**Review by Kiransukumaran — CHANGES_REQUESTED** [review]

File: src/services/accountService.ts (hunk 6 above)
Comment (anujakiran) [reply src/services/accountService.ts:28]: Please raise a new ticket to add logs for the entire project. Will address this later

**Review by Kiransukumaran — APPROVED** [review]

**Pull request merged**